File: generate_icons.py
#!/usr/bin/env python
import os
from PIL import Image

# Ensure the images directory exists
if not os.path.exists('images'):
    os.makedirs('images')

def resize_icon(input_image, size):
    """Resize the input image to the specified size and save it as an icon."""
    print(f"Creating {size}x{size} icon...")
    
    # Open the source image
    try:
        with Image.open(input_image) as img:
            # Resize the image, maintaining aspect ratio
            img = img.resize((size, size), Image.LANCZOS)
            
            # Save the resized image
            output_file = f"images/icon{size}.png"
            img.save(output_file)
            print(f"Created {output_file}")
    except Exception as e:
        print(f"Error creating icon: {e}")

# Main execution
try:
    # Source image path
    source_icon = "images/books_icon.png"
    
    # Check if source image exists
    if not os.path.exists(source_icon):
        print(f"Error: Source image {source_icon} not found!")
        exit(1)
        
    # Create icons in all required sizes
    resize_icon(source_icon, 16)
    resize_icon(source_icon, 48)
    resize_icon(source_icon, 128)
    
    print("All icons created successfully!")
except Exception as e:
    print(f"Error: {e}") 

File: generate_icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

// Make sure the images directory exists
if (!fs.existsSync('./images')) {
  fs.mkdirSync('./images');
}

// Function to create an icon
function createIcon(size) {
  console.log(`Creating ${size}x${size} icon...`);
  
  // Create canvas with the given size
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Draw blue background
  ctx.fillStyle = '#4285f4';
  ctx.fillRect(0, 0, size, size);
  
  // Draw white "APS" text
  ctx.fillStyle = 'white';
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  
  // Set font size based on icon size
  const fontSize = Math.floor(size * 0.5);
  ctx.font = `bold ${fontSize}px Arial, sans-serif`;
  
  // Draw text
  ctx.fillText('APS', size/2, size/2);
  
  // Save to PNG file
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(`./images/icon${size}.png`, buffer);
  
  console.log(`Created images/icon${size}.png`);
}

// Create icons of all required sizes
try {
  createIcon(16);
  createIcon(48);
  createIcon(128);
  console.log('All icons generated successfully!');
} catch (err) {
  console.error('Error generating icons:', err);
} 